EIPS/eip-6493.md

```diff
@@ -0,0 +1,226 @@
+---
+eip: 6493
+title: SSZ Transaction Signature Scheme
+description: Signature scheme for SSZ transactions
+author: Etan Kissling (@etan-status), Matt Garnett (@lightclient), Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/eip-6493-ssz-transaction-signature-scheme/13050
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-02-24
+requires: 155, 2124, 2718
+---
+
+## Abstract
+
+This EIP defines a signature scheme for [Simple Serialize (SSZ)](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/ssz/simple-serialize.md) encoded transactions.
+
+## Motivation
+
+Existing [EIP-2718](./eip-2718.md) transaction types are encoded in the RLP format, and then hashed using keccak256 for signing. For new transaction types that are encoded in the SSZ format, it is ideomatic to base their signature hash and their public identifier on `hash_tree_root` instead.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### SSZ transaction schema
+
+For each SSZ transaction type, its specification defines an SSZ type that contains all transaction data to be signed.
+
+```python
+class XyzTransaction(View):
+    ...
+```
+
+Each specification also defines an SSZ type that represents the transaction's signature.
+
+```python
+class XyzSignature(View):
+    ...
+```
+
+The signed transaction combines the unsigned transaction data with its signature.
+
+```python
+class XyzSignedTransaction(Container):
+    message: XyzTransaction
+    signature: XyzSignature
+```
+
+The unique transaction identifier is defined as this signed container's SSZ `hash_tree_root`.
+
+When representing the transaction as part of the [execution block header's `txs-root`](https://github.com/ethereum/devp2p/blob/bd17dac4228c69b6379644355f373669f74952cd/caps/eth.md#block-encoding-and-validity) Merkle-Patricia Trie, that unique transaction identifier is embedded instead of the transaction's raw network representation.
```

Comment:
```suggestion
When representing such a  transaction as part of the [execution block header's `txs-root`](https://github.com/ethereum/devp2p/blob/bd17dac4228c69b6379644355f373669f74952cd/caps/eth.md#block-encoding-and-validity) Merkle-Patricia Trie, this unique transaction identifier is embedded instead of the transaction's raw network representation.
```

Reply:
Typo fix:
```
When representing such a transaction as part of the [execution block header's `txs-root`](https://github.com/ethereum/devp2p/blob/bd17dac4228c69b6379644355f373669f74952cd/caps/eth.md#block-encoding-and-validity) Merkle-Patricia Trie, this unique transaction identifier is embedded instead of the transaction's raw network representation.
```